flare/gp_algebra.py

```diff
@@ -504,4 +505,9 @@ def get_neg_like_grad(hyps, training_data: list,
     if output is not None:
         output.write_hyps(None, hyps, None, like, like_grad, name="hyps")
 
+    if print_progress:
```

Comment:
maybe merge this condition block to the one above and make it "elif print_progress" ? just to reduce redundancy

Reply:
Hmm I think we should leave open the option to do both, i.e. to both print to console to check the progress in real time, and to also record the progress in an output file so that it's stored.